lms/static/scripts/frontend_apps/utils/fetch.ts

```diff
@@ -109,14 +116,17 @@ export function useFetch<T = unknown>(
               retry: doFetch,
             });
           }
-        });
+        })
+        .finally(() => globalLoading?.finishLoading(key));
     };
     doFetch();
 
     return () => {
       // Cancel in-flight request if query changes or component is unmounted.
       controller.abort();
     };
+    // We don't want this effect to run
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [key]);
```

Comment:
`react-hooks/exhaustive-deps` wants the `globalLoading` state to be added as a dependency of the hook. However, that causes a double fetch, because the reference changes when it transitions from `initialLoadingInProgress = true` to `initialLoadingInProgress = false`.

I'm trying to figure this out, as I think there has to be a way.

Reply:
Ok, found the solution.